code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -38,6 +38,51 @@
 		if((m != R.module_state_1) && (m != R.module_state_2) && (m != R.module_state_3))
 			. += m
 
+/obj/item/weapon/robot_module/proc/add_module(var/obj/item/I)
+	if(istype(I, /obj/item/stack))
+		var/obj/item/stack/S = I
+
+		if(is_type_in_list(S, list(/obj/item/stack/sheet/metal, /obj/item/stack/rods, /obj/item/stack/tile/plasteel)))
+			var/datum/robot_energy_storage/metal/metstore = locate(/datum/robot_energy_storage/metal) in storages
+			if(!metstore)
+				metstore = new /datum/robot_energy_storage/metal(src)
+
+			if(S.materials[MAT_METAL])
+				S.cost = S.materials[MAT_METAL] / 4

```

Comment:
Yeah, as long as the reciprocal is decimal expressible (1.1 is decimal expressible because it terminates; 0.111... is not decimal expressible because it does not terminate), multiplying should be used. 
Note that this only works if we know the divisor; if we have a variable as the divisor then we must use the division.